config/routes.rb

```diff
@@ -16,7 +16,9 @@
   resources :comments, :only => [:destroy, :edit, :update]
   resources :users, :only => [:show, :edit, :update]
   namespace :admin do
-    resources :users
+    resources :users do
+      resources :badges, :only => [:create, :destroy]
```

Comment:
This hooks up actions in the `badges_controntroller`, yet you made your changes to the `users_controller`.  I think you might want to move the logic in that controller into a new controller you generate.